cmake/extra_flags.cmake

```diff
@@ -1,10 +1,10 @@
 # SPDX-License-Identifier: Apache-2.0
 
-separate_arguments(EXTRA_CPPFLAGS_AS_LIST UNIX_COMMAND ${EXTRA_CPPFLAGS})
-separate_arguments(EXTRA_LDFLAGS_AS_LIST UNIX_COMMAND  ${EXTRA_LDFLAGS})
-separate_arguments(EXTRA_CFLAGS_AS_LIST   UNIX_COMMAND ${EXTRA_CFLAGS})
-separate_arguments(EXTRA_CXXFLAGS_AS_LIST UNIX_COMMAND ${EXTRA_CXXFLAGS})
-separate_arguments(EXTRA_AFLAGS_AS_LIST   UNIX_COMMAND ${EXTRA_AFLAGS})
+string(REPLACE " " ";" EXTRA_CPPFLAGS_AS_LIST "${EXTRA_CPPFLAGS}")
```

Comment:
will this work if we have:
```
-DXXX="--arg \"foo bar\" --arg2..."
```

(if it's possible)?

Reply:
that's actually a good point.
The space inside those quotes will also be replaced, something we don't want in this particular case.

Will take an extra look at this.

Reply:
@gmarull finally found the time to come back to this.

This is now fixed by still supporting the space separated lists, that requires spaces within arguments to have extra quoting.

Note how that is not needed with proper CMake lists, as the `;` separates the values like:
```
-DXXX="--arg;foo bar;--arg2"
```

Of course this still works for backward compatibility:
```
-DXXX="--arg \"foo bar\" --arg2"
```